pkg/mimir/mimir.go

```diff
@@ -785,11 +798,12 @@ func (t *Mimir) Run() error {
 
 	sm.AddListener(services.NewManagerListener(healthy, stopped, serviceFailed))
 
-	// Setup signal handler. If signal arrives, we stop the manager, which stops all the services.
-	handler := signals.NewHandler(t.Server.Log)
+	// Setup signal handler. If a signal arrives, we mark ourselves as "not ready", disable HTTP
+	// keep-alives, optionally wait some amount of time, and then stop the manager which stops
+	// all dependent the services.
```

Comment:
All this happens in signal receiver, and not here. Let's move the logic here to keep the comment.

"which stops all dependent the services" => "which stops all the services".